conans/client/cmd/export.py

```diff
@@ -32,13 +32,25 @@ def cmd_export(conanfile_path, name, version, user, channel, keep_source,
     conan_linter(conanfile_path, output)
     conanfile = _load_export_conanfile(conanfile_path, output, name, version)
     conan_ref = ConanFileReference(conanfile.name, conanfile.version, user, channel)
+    if conan_ref.revision:  # Could be improved modifying the recipe?
+        raise ConanException("Specify the revision in a recipe field '_revision_'")
+    elif conanfile._revision_:
+        try:
+            int(conanfile._revision_)
+        except ValueError:
+            raise ConanException("Specify an integer revision")
+        conan_ref = ConanFileReference(conanfile.name, conanfile.version, user, channel, conanfile._revision_)
+
     conan_ref_str = str(conan_ref)
     # Maybe a platform check could be added, but depends on disk partition
     refs = DiskSearchManager(client_cache).search_recipes(conan_ref_str, ignorecase=True)
     if refs and conan_ref not in refs:
         raise ConanException("Cannot export package with same name but different case\n"
                              "You exported '%s' but already existing '%s'"
                              % (conan_ref_str, " ".join(str(s) for s in refs)))
+
+    # FIXME: Check exporting without revision where there are revisions?
```

Comment:
Pending validations? Like mixing with or without (youuuu) revisions in the local cache.